src/lib.rs

```diff
@@ -111,18 +113,54 @@ pub fn parse(raw_languages: &str) -> Vec<String> {
 /// ```
 pub fn intersection(raw_languages: &str, supported_languages: &[&str]) -> Vec<String> {
     let user_languages = parse(raw_languages);
-
     user_languages
         .into_iter()
         .filter(|l| supported_languages.contains(&l.as_str()))
         .collect()
 }
 
+/// Compare an Accept-Language header value with your application's supported languages,
+/// which MUST be in alphabetical order, to find the common languages that could be presented
+/// to a user. Executes roughly 25% faster.
+///
+/// # Example
+///
+/// ```
+/// use accept_language::intersection_fast;
+///
+/// let common_languages = intersection_fast("en-US, en-GB;q=0.5", &["de", "en-GB", "en-US"]);
+/// ```
+pub fn intersection_fast(raw_languages: &str, supported_languages: &[&str]) -> Vec<String> {
```

Comment:
I wonder if `intersection_ordered` might be a better name? That's the important part, and I think adding `ordered` to the name would cause people to pay more attention to the docs than `fast` would

Reply:
Indeed, better name it after the method than the result. Also the quality member in the struct could be compacted to a `f32` instead of a `f64`, don't you think?

Reply:
Yeah, that's probably fine!